python_modules/libraries/dagster-airlift/dagster_airlift/core/monitoring_job/event_stream.py

```diff
@@ -260,6 +266,43 @@ def _process_completed_runs(
             break
 
 
+async def _retrieve_logs_for_task_instance(
+    context: OpExecutionContext,
+    airflow_instance: AirflowInstance,
+    task_instance: TaskInstance,
+) -> TaskInstanceCompleted:
+    logs = airflow_instance.get_task_instance_logs(
```

Comment:
is the io happening here?

Reply:
that's right yea

Reply:
this should be `await` all the way [through](https://github.com/dagster-io/dagster/blob/a81d80a18791ac01786f5d0996ed2291e8ea6377/python_modules/libraries/dagster-airlift/dagster_airlift/core/airflow_instance.py#L553) to the actual io. otherwise the sync-io will just block other asyncio tasks in this loop.